src/RESTController.js

```diff
@@ -158,15 +158,32 @@ const RESTController = {
       for (const key in customHeaders) {
         headers[key] = customHeaders[key];
       }
-      xhr.onprogress = (event) => {
-        if(options && typeof options.progress === 'function') {
+
+      function handleProgress(type, event) {
+        if (options && typeof options.progress === 'function') {
           if (event.lengthComputable) {
-            options.progress(event.loaded / event.total, event.loaded, event.total);
+            options.progress(
+              type,
+              event.loaded / event.total,
+              event.loaded,
+              event.total
+            );
           } else {
-            options.progress(null);
+            options.progress(type, null);
           }
         }
+      }
+
+      xhr.onprogress = (event) => {
+        handleProgress('download', event);
       };
+
+      if (xhr.upload) {
```

Comment:
Oops I forgot why this was removed. The type parameter can be removed here and we should be good to go.